samples/animate-images-with-image-overlay/src/main/java/com/esri/arcgismaps/sample/animateimageswithimageoverlay/components/AnimateImagesWithImageOverlayViewModel.kt

```diff
@@ -0,0 +1,185 @@
+        width = 15.0958,
+        height = -14.3770
+    )
+
+    // Create a scene with the dark gray basemap and elevation source
+    val arcGISScene by mutableStateOf(ArcGISScene(BasemapStyle.ArcGISDarkGray).apply {
+        // Create a camera, looking at the pacific southwest sector
+        val observationPoint = Point(-116.621, 24.7773, 856977.0)
+        val camera = Camera(observationPoint, 353.994, 48.5495, 0.0)
+        initialViewpoint = Viewpoint(pacificSouthwestEnvelope, camera)
+    })
+
+    // Create a message dialog view model for handling error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Keep track of the list of image frames added in cache
+    private var imageFrames = mutableListOf<ImageFrame>()
+
+    // Keep track of the image frame currently in view
+    private var imageFrameIndex = 0
+
+    // Timer task to customize frame rates
+    private var timer: Timer? = null
+
+    var imageOverlay = ImageOverlay()
+        private set
+
+    init {
+
+        // Get the image files from local storage as an unordered list
+        (File(filePath).listFiles())?.let { imageFiles ->
+            // Sort the list of image files
+            Arrays.sort(imageFiles)
+            imageFiles.forEach { file ->
+                // Create an image with the given path and use it to create an image frame
+                val imageFrame = ImageFrame(file.path, pacificSouthwestEnvelope)
+                imageFrames.add(imageFrame)
+            }
+            // Set the initial image frame to image overlay
+            imageOverlay.imageFrame = imageFrames[imageFrameIndex]
+        }
+
+        viewModelScope.launch {
+            arcGISScene.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }
+        }
+    }
+
+    /**
+     * Create a new image frame from the image at the current index and add it to the image overlay.
+     */
+    private fun addNextImageFrameToImageOverlay() {
+        // Set image frame to image overlay
+        imageOverlay.imageFrame = imageFrames[imageFrameIndex]
+        // Increment the index to keep track of which image to load next
+        imageFrameIndex++
+        // Reset index once all files have been loaded
+        if (imageFrameIndex == imageFrames.size)
+            imageFrameIndex = 0
+    }
+
+    /**
+     * Create a new timer for the given period which repeatedly calls [addNextImageFrameToImageOverlay]..
+     */
+    private fun createNewTimer() {
+        val period = when (fps) {
+            60 -> 17 // 1000ms/17 = 60 fps
+            30 -> 33 // 1000ms/33 = 30 fps
+            15 -> 67 // 1000ms/67 = 15 fps
+            else -> 0
+        }
+        timer?.cancel()
+        timer = fixedRateTimer("Image overlay timer", period = period.toLong()) {
+            addNextImageFrameToImageOverlay()
+        }
```

Comment:
`fps` could be part of an internal flow that creates a new `fixedRateTimer` on each update. This would allow us to change the fps of the animation in real-time.